lib/fetcher/extractionPool.ts

```diff
@@ -0,0 +1,127 @@
+import { Worker } from "node:worker_threads";
+import { LDESInfo } from "./memberManager";
+import type { IncomingMessage, OutgoingMessage } from "./extractionWorker";
+import { Parser, Quad_Object, Writer } from "n3";
+import { Quad, Term } from "@rdfjs/types";
+import { DataFactory, DefaultGraph } from "rdf-data-factory";
+import { RdfStore } from "rdf-stores";
+
+const { namedNode, quad } = new DataFactory();
+
+type Input = {
+    members: Term[];
+    quads: RdfStore;
+    handleMember: (quads: Quad[], id: Term) => void;
+    done: () => void;
+};
+
+type State = {
+    worker: Worker;
+    state: "busy" | "idle";
+    handleMember: (quads: Quad[], id: Term) => void;
+    done: () => void;
+};
+
+export class Pool {
+    private readonly workers: State[];
+    private readonly queue: Input[] = [];
+    constructor(info: LDESInfo, workerCount = 1) {
+        console.log({ workerCount });
```

Comment:
Use `logger` instead of `console.log` or remove if it was for dev-only purposes.